BlockSettleUILib/UiUtils.cpp

```diff
@@ -612,14 +612,21 @@ QString UiUtils::NormalizeString(const QString& value)
 
 QValidator::State UiUtils::ValidateDoubleString(QString &input, int &pos, const int decimals)
 {
-   if (input.isEmpty()) {
-      return QValidator::Acceptable;
-   }
-
    static const QChar defaultDecimalsSeparatorChar = QLatin1Char('.');
 
    QString tempCopy = UiUtils::NormalizeString(input);
 
+   QStringList list = tempCopy.split(defaultDecimalsSeparatorChar);
+   if (list.size() == 2 && list[1].length() > decimals && pos == input.length()) {
+      list[1].resize(decimals);
+      input = list[0] + QLatin1Char('.') + list[1];
```

Comment:
why are you updating input? I don't think that it is expected for that string to change

Reply:
From https://doc.qt.io/qt-5/qvalidator.html#validate
> This virtual function returns Invalid if input is invalid according to this validator's rules, Intermediate if it is likely that a little more editing will make the input acceptable (e.g. the user types "4" into a widget which accepts integers between 10 and 99), and Acceptable if the input is valid.
The function can change both input and pos (the cursor position) if required.